fairlearn/metrics/_metric_frame.py

```diff
@@ -0,0 +1,552 @@
+    feature(s). The name 'control features' refers to the statistical practice
+    of 'controlling' for a variable.
+
+    Parameters
+    ----------
+    metric : callable or dict
+        The underlying metric functions which are to be calculated. This
+        can either be a single metric function or a dictionary of functions.
+        These functions must be callable as
+        ``fn(y_true, y_pred, **sample_params)``.
+        If there are any other arguments required (such as ``beta`` for
+        :func:`sklearn.metrics.fbeta_score`) then
+        :func:`functools.partial` must be used.
+
+        **Note** that the values returned by various members of the class change
+        based on whether this argument is a callable or a dictionary of
+        callables. This distinction remains *even if* the dictionary only
+        contains a single entry.
+
+    y_true : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The true values.
+
+    y_pred : List, pandas.Series, numpy.ndarray, pandas.DataFrame
+        The predicted values.
+
+    sensitive_features : List, pandas.Series, dict of 1d arrays, numpy.ndarray, pandas.DataFrame
+        The sensitive features which should be used to create the subgroups.
+        At least one sensitive feature must be provided.
+        All names (whether on pandas objects or dictionary keys) must be strings.
+        We also forbid DataFrames with column names of ``None``.
+        For cases where no names are provided we generate names ``sensitive_feature_[n]``.
+
+    control_features : List, pandas.Series, dict of 1d arrays, numpy.ndarray, pandas.DataFrame
+        Control features are similar to sensitive features, in that they
+        divide the input data into subgroups.
+        Unlike the sensitive features, aggregations are not performed
+        across the control features - for example, the ``overall`` property
+        will have one value for each subgroup in the control feature(s),
+        rather than a single value for the entire data set.
+        Control features can be specified similarly to the sensitive features.
+        However, their default names (if none can be identified in the
+        input values) are of the format ``control_feature_[n]``.
+
+        **Note** the types returned by members of the class vary based on whether
+        control features are present.
+
+    sample_params : dict
+        Parameters for the metric function(s). If there is only one metric function,
+        then this is a dictionary of strings and array_likes, which are split
+        alongside the ``y_true`` and ``y_pred`` arrays, and passed to the metric function.
+        If there are multiple metric functions (passed as a dictionary), then this is
+        a nested dictionary, with the first set of string keys identifying the
+        metric function name, with the values being the string-to-array_like dictionaries.
```

Comment:
just flagging `array_like` in case you'd like to change it, but i'm actually fine keeping as is.

Reply:
Switched anyway.